### PR TITLE
Fix returned File node order when using arrays

### PR DESCRIPTION
**Pull request by sbuys**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,9 +14,11 @@
...
     "@types/node": "^12.11.1",
-    "typescript": "^3.6.4"
-  },
-  "peerDependencies": {
-    "gatsby": "^2.16"
+    "@types/react": "^17.0.2",
+    "@types/react-dom": "^17.0.1",
+    "gatsby": "^2.16",
+    "react": "^16.0",
+    "react-dom": "^16.0",
```
Comment (d4rekanguok) [inline package.json]: I think we're not doing anything with react & react-dom, right?

File: .prettierrc [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+  "semi": false,
+  "singleQuote": true,
+  "trailingComma": "es5"
+}
```
Comment (d4rekanguok) [inline .prettierrc:5]: Thanks for this! Didn't realized I don't have prettier setup in this repo

File: src/gatsby-node.ts [hunk 3]
```diff
@@ -64,31 +68,41 @@ const createResolve: CreateResolve = ({
     return fileNodes[0]
   }
 
-  return fileNodes
+  // Return in original order
+  const fileNodesOrdered = filePaths.flatMap((filepath) => {
```
Comment (d4rekanguok) [inline src/gatsby-node.ts]: Nice! But why `flatMap`  vs. `map`? I thought `find` returns the item itself & not wrapped in an array

File: src/gatsby-node.ts [hunk 4]
```diff
@@ -64,31 +68,41 @@ const createResolve: CreateResolve = ({
...
+    return fileNodes.find((node: any) => {
+      console.log(node.absolutePath, filepath)
```
Comment (d4rekanguok) [inline src/gatsby-node.ts]: a stray console.log!

**Review by d4rekanguok — COMMENTED** [review]
Thanks so much @sbuys! It looks like there're a few unused deps left in (react, react-dom) and a stray line of console.log — otherwise it looks good to me.

File: src/gatsby-node.ts (hunk 3 above)
Comment (sbuys) [reply src/gatsby-node.ts]: You're right. That context switch from Swift will getcha `flatMap / compactMap`. 

It shouldn't ever happen in this use case, but `find` can return `undefined`. Should remove any potential dirty values from the array.

**Comment by d4rekanguok** [comment]
@sbuys Ok, I see why you added `gatsby` & `react`, `react-dom` to `devDependencies` — I'm so sorry that dealing with typescript & gatsby plugin system is such a pain, and thank you for making the time to figure it all out!

In my experience, adding those deps to a gatsby plugin sometimes cause unexpected module resolution issues. So I've been biting the bullet, and just... install `gatsby` then manually removed it from my package.json.

The changes has been merged & published in v1.2.3.

**Comment by sbuys** [comment]
@d4rekanguok Thanks for moving so quickly on this and for maintaining this plugin. Super helpful :).

**Pull request merged**